ledfx/devices/ddp.py

```diff
@@ -62,8 +63,14 @@ def flush(self, data):
                 data,
                 self.frame_count,
             )
+            self.connection_warning = False
         except AttributeError:
             self.activate()
+        except OSError as e:
+            # print warning only once until it clears
+            if not self.connection_warning:
+                _LOGGER.error("Error connecting by DDP: %s", e)
+                self.connection_warning = True
```

Comment:
Interesting advice. If wanted I can add logging about a cleared warning. Also adding `{self.config['name']}` to the logging might be helpful?
And should the log level here be `error` or `warning`? What should the "cleared" level be, if wanted?

Reply:
Thanks for moving the logging level down to info. When logging is at error level, it will hit the sentry back end, and we would not want to see this transient issue being sent to sentry. We are aggressively cleaning it down, so we can address the signal in the noise.

Meanwhile prior to this code change, when I locally power off a WLED endpoint on DDP, and later power it back up and see it recover, I don't get any logging. So is the case you wanted to explicitly add "something" without spamming?

Reply:
The level is now `warning`.

Seems the crash may depend on the OS and router setup perhaps. From what I see the router notifies (ICMP) that the device is unreachable and then I get an `OSError: [Errno 65] No route to host` exception in Python. It seems to require the device to drop from ARP resolution tables, so you might need to disconnect for a longer time?

Reply:
hmm, just running the branch, windows 11 and I don't hit an OSError, I will put a trap all to see if try is firing at all

Reply:
As DDP is UDP based, I would not expect any OS level awareness of the loss of an endpoint. There is no managed session, we are just throwing packet down a dark hole, and hoping they arrive. Are you able to reproduce this OS Error scenario in your setup? Can you get your debug to actually fire?

Reply:
Trying to understand where you are able to trigger the failure, I suspect its more in the routing table / DNS lookup or similar. I appreciate that you are triggering the issue, I would just like to understand a little more, as I can't stimulate it directly. For example, do you have device names or IP in your IP Address field

Reply:
Here is the crash without catching the exception, a few seconds after startup:
```
Exception in thread Virtual: my-ddp-dev:
Traceback (most recent call last):
  File "lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/virtuals.py", line 507, in thread_function
Exception in thread Virtual: balken:
Traceback (most recent call last):
  File "lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.flush()
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/virtuals.py", line 706, in flush
    self.run()
  File "lib/python3.11/threading.py", line 982, in run
    device.update_pixels(self.id, data)
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/devices/__init__.py", line 152, in update_pixels
    self._target(*self._args, **self._kwargs)
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/virtuals.py", line 507, in thread_function
    self.flush(frame)
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/devices/ddp.py", line 59, in flush
    self.flush()
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/virtuals.py", line 706, in flush
    DDPDevice.send_out(
    device.update_pixels(self.id, data)
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/devices/ddp.py", line 88, in send_out
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/devices/__init__.py", line 152, in update_pixels
    self.flush(frame)
    DDPDevice.send_packet(
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/devices/ddp.py", line 59, in flush
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/devices/ddp.py", line 114, in send_packet
    DDPDevice.send_out(
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/devices/ddp.py", line 88, in send_out
    sock.sendto(
OSError: [Errno 65] No route to host
    DDPDevice.send_packet(
  File "ledfx_venv/lib/python3.11/site-packages/ledfx/devices/ddp.py", line 114, in send_packet
    sock.sendto(
OSError: [Errno 65] No route to host
```

Reply:
It takes some time for, I guess, ARP resolution to fail. Minimal repro:
```
import socket
from time import sleep
sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
for n in range(100):
    sock.sendto(b'MESSAGE', ("192.168.1.2", 4048))
    sleep(0.1)
```
(make sure `192.168.1.2` is local but not reachable) then this runs for ~3.5 secs and one might get
```
7
7
[…]
7
7
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
OSError: [Errno 65] No route to host
```

Reply:
Tried it on a few OS now and it seems specific to BSD/MacOS.

Reply:
OK, if you bump that down to warning, then I would be happy for this to go in. I can't test locally, don't have MacOS but will trust that you test exhaustively :-)

Reply:
I see you have moved to warning, thankyou. @shauneccles I am good with this